### ISSUE TITLE
`SIM102` false positive when using walrus operator

### ISSUE THREAD

**Issue by adam-sikora** [astral-sh/ruff#12363]
The following code snippet does not pass validation for rule `SIM102` (Use a single if statement instead of nested if statements)
```
foo = {"a": 1}
if bar := foo["a"]:
    if bar == 1:
        print(bar)
```
Check command used: `ruff check --select="SIM102" example.py --isolated` (ruff version: `0.5.2`)

However the if statements cannot be combined as the rule suggests as
```
foo = {"a": 1}
if bar := foo["a"] and bar == 1:
    print(bar)
```
would lead to: `NameError: name 'bar' is not defined`

**Comment by charliermarsh** [comment]
I think it does work as:

```python
foo = {"a": 1}
if (bar := foo["a"]) and bar == 1:
    print(bar)
```

**Comment by adam-sikora** [comment]
Yeah, my bad for forgetting the correct operator priority. Thanks for the clarification
